src/gpuarray_array.c

```diff
@@ -1008,15 +1009,18 @@ int GpuArray_concatenate(GpuArray *r, const GpuArray **as, size_t n,
 
   res_off = r->offset;
   res_dims = r->dimensions;
+  res_flags = r->flags;
+  r->flags &= ~(GA_C_CONTIGUOUS|GA_F_CONTIGUOUS);
   for (i = 0; i < n; i++) {
     r->dimensions = as[i]->dimensions;
-    err = GpuArray_move(r, as[i]);
+    err = ga_extcopy(r, as[i]);
```

Comment:
It is a shortcut and should be a bit faster since we avoid the checks in GpuArray_move().  I used move as an attempt to use memcopy whenever possible, but this broke a lot of cases so I rolled back to always extcopy.

We can attempt to re-introduce the memcopy later if it ever becomes a problem.